src/main/java/org/jenkinsci/test/acceptance/plugins/ssh_slaves/SshSlaveLauncher.java

```diff
@@ -30,7 +31,24 @@ public SshSlaveLauncher(PageObject context, String path) {
     }
 
     public SshCredentialDialog addCredential() {
-        self().findElement(by.button("Add")).click();
+
+        if (getElement(by.xpath("//span[contains(@class,'credentials-add')]")) == null) {
+
+            String providerXpathExpr = "//div[contains(@class,'credentials-add-menu-items')]"
+                    + "/div[@class='bd']/ul[@class='first-of-type']/li[contains(@class, 'yuimenuitem')]"
+                    + "/span[contains(@class,'yuimenuitemlabel') and contains(text(), 'Jenkins')]";
+
+            self().findElement(by.xpath("//button[@class='credentials-add-menu']")).click();
+
+            for (WebElement menuItem : self().findElements(by.xpath(providerXpathExpr))) {
+                if (menuItem.isDisplayed()) {
+                    menuItem.click();
+                    break;
```

Comment:
Ordering shouldn't matter - that's why we're looking for the `"Jenkins"` menu item, not just the first one.